internal/persistence/postgres/txhistory.go

```diff
@@ -95,18 +95,17 @@ func (p *sqlPersistence) ListTransactionHistory(ctx context.Context, txID string
 	return p.txHistory.GetMany(ctx, filter.Condition(filter.Builder().Eq("transaction", txID)))
 }
 
-func (p *sqlPersistence) AddSubStatusAction(ctx context.Context, txID string, subStatus apitypes.TxSubStatus, action apitypes.TxAction, info *fftypes.JSONAny, errInfo *fftypes.JSONAny) error {
+func (p *sqlPersistence) AddSubStatusAction(ctx context.Context, txID string, subStatus apitypes.TxSubStatus, action apitypes.TxAction, info *fftypes.JSONAny, errInfo *fftypes.JSONAny, t *fftypes.FFTime) error {
 	// Dispatch to TX writer
-	now := fftypes.Now()
 	op := newTransactionOperation(txID)
 	op.historyRecord = &apitypes.TXHistoryRecord{
 		ID:            fftypes.NewUUID(),
 		TransactionID: txID,
 		SubStatus:     subStatus,
 		TxHistoryActionEntry: apitypes.TxHistoryActionEntry{
 			OccurrenceCount: 1,
-			Time:            now,
-			LastOccurrence:  now,
+			Time:            t,
+			LastOccurrence:  t,
```

Comment:
<img width="668" alt="image" src="https://github.com/hyperledger/firefly-transaction-manager/assets/5425125/863bebb7-864f-470d-8575-b0d1c6ef79d5">
the line below also needs changing.

We used to have the idea of occurrence count, but it was abandoned during the recent refactor, so it's OK to set all three timestamps to the same value for now

Reply:
roger that, i will make changes